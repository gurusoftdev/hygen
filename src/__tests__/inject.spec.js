const inject = require('../inject')
const gemfile = `
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    `
describe('inject', () => {
  it('before', () => {
    expect(
      inject(
        {
          attributes: {
            before: "gem 'rails'"
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('after', () => {
    expect(
      inject(
        {
          attributes: {
            after: "gem 'rails'"
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('prepend', () => {
    expect(
      inject(
        {
          attributes: {
            prepend: true
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('append', () => {
    expect(
      inject(
        {
          attributes: {
            append: true
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('at_index', () => {
    expect(
      inject(
        {
          attributes: {
            at_line: 2
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
  it('skip_if', () => {
    expect(
      inject(
        {
          attributes: {
            skip_if: 'source',
            after: "gem 'rails'"
          },
          body: "    gem 'kamikaze' # added by hygen"
        },
        gemfile
      )
    ).toMatchSnapshot()
  })
})
