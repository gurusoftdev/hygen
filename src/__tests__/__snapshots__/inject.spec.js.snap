// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inject after 1`] = `
Object {
  "attributes": Object {
    "after": "gem 'rails'",
  },
  "body": "
    source 'http://rubygems.org'
    gem 'rails'
    gem 'kamikaze' # added by hygen
    gem 'nokogiri'
    gem 'httparty'

    ",
}
`;

exports[`inject append 1`] = `
Object {
  "attributes": Object {
    "append": true,
  },
  "body": "
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    gem 'kamikaze' # added by hygen
    ",
}
`;

exports[`inject at_index 1`] = `
Object {
  "attributes": Object {
    "at_line": 2,
  },
  "body": "
    source 'http://rubygems.org'
    gem 'kamikaze' # added by hygen
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    ",
}
`;

exports[`inject before 1`] = `
Object {
  "attributes": Object {
    "before": "gem 'rails'",
  },
  "body": "
    source 'http://rubygems.org'
    gem 'kamikaze' # added by hygen
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    ",
}
`;

exports[`inject prepend 1`] = `
Object {
  "attributes": Object {
    "prepend": true,
  },
  "body": "
    gem 'kamikaze' # added by hygen
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    ",
}
`;

exports[`inject skip_if 1`] = `
Object {
  "attributes": Object {
    "after": "gem 'rails'",
    "skip_if": "source",
  },
  "body": "
    source 'http://rubygems.org'
    gem 'rails'
    gem 'nokogiri'
    gem 'httparty'

    ",
}
`;
